Extension/translations_auto_pr.js

```diff
@@ -0,0 +1,126 @@
+}
+
+// When invoked on build server, we should already be in a repo freshly synced to margeTo branch
+
+if (hasAnyChanges()) {
+    console.log(`Changes already present in this repo!  This script is intended to be run against a freshly synced ${mergeTo} branch!`);
+    return;
+}
+
+function sleep(ms) {
+    var unixtime_ms = new Date().getTime();
+    while(new Date().getTime() < unixtime_ms + ms) {}
+}
+
+console.log("This script is potentially DESTRUCTIVE!  Cancel now, or it will proceeed in 10 seconds.");
+sleep(10000);
+
+console.log("Looking for latest localization drop");
+let latestTxt = fs.readFileSync("\\\\simpleloc\\drops\\Drops\\vscode-extensions_2432\\Latest.txt")
+let exp = /Build No:\s*(.*)(?!=\n)/gm;
+let match = exp.exec(latestTxt.toString());
+let versionString = match[1];
+
+console.log("Copying XLF files to vscode-translations-import directory");
+let rootSourcePath = `\\\\simpleloc\\drops\\Drops\\vscode-extensions_2432\\${versionString}\\Localization\\locdrop\\bin`; 
+let directories = fs.readdirSync(rootSourcePath);
+directories.forEach(folderName => {
+    let sourcePath = `${rootSourcePath}\\${folderName}\\Projects\\Src\\vscode-extensions\\vscode-${locProjectName}.${folderName}.xlf`;
+    let destinationPath = `../vscode-translations-import/${folderName}/vscode-extensions/vscode-${locProjectName}.xlf`;
+    console.log(`Copying "${sourcePath}$" to "${destinationPath}"`);
+    fs.copySync(sourcePath, destinationPath);
+});
+
+console.log("Import translations into i18n directory");
+cp.execSync("npm run translations-import");
+
+if (!hasAnyChanges()) {
+    console.log("No changes detected");
+    return;
+}
+
+console.log("Changes detected");
+
+// Check out local branch
+console.log(`Creating local branch for changes (git checkout -b ${branchName})`);
+cp.execSync('git checkout -b localization');
+
+// Add changed files.
+console.log("Adding changed file (git add .)");
+cp.execSync('git add .');
+
+// Commit changes files.
+console.log(`Commiting changes (git commit -m "${commitComment}")`);
+cp.execSync(`git commit -m "${commitComment}"`);
+
+console.log(`Configuring git with permission to push and to create pull requests (git remote remove origin && git remote add origin https://${authUser}:${authToken}@github.com/${repoOwner}/${repoName}.git`);
+cp.execSync('git remote remove origin');
+cp.execSync(`git remote add origin https://${authUser}:${authToken}@github.com/${repoOwner}/${repoName}.git`);
+
+console.log(`pushing to remove branch (git push -f origin ${branchName})`);
+cp.execSync(`git push -f origin ${branchName}`);
+
+console.log("Checking if there is already a pull request...");
+const octokit = new Octokit({auth: {
+    username: authUser,
+    password: authToken}
+});
+octokit.pulls.list({ owner: repoOwner, repo: repoName }).then(({data}) => {
+    let alreadyHasPullRequest = false;
+    if (data) {
+        data.forEach((pr) => {
+            alreadyHasPullRequest |= pr.title === pullRequestTitle;
```

Comment:
Maybe using alreadyHasPullRequest = alreadyHasPullRequest || pr.title === pullRequestTitle would be "safer"?